src/methods/SANA.cpp

```diff
@@ -1208,41 +1196,34 @@ void SANA::performSwap(int type) {
     double newCurrentScore = 0;
     bool makeChange = scoreComparison(newAligEdges, inducedEdges, newTCSum, newLocalScoreSum, newWecSum, newNcSum, newCurrentScore, newEwecSum, newSquaredAligEdges, newExposedEdgesNumer);
 
-    if(wasBadMove && !makeChange)
-    {
-        #ifdef CORES
-			// Statistics on the emerging core alignment.
-			// only update pBad if it's nonzero; re-use previous nonzero pBad if the current one is zero.
-			static double pBad;
-			if(trueAcceptingProbability()>0) pBad = trueAcceptingProbability();
+    #ifdef CORES
+        // Statistics on the emerging core alignment.
+        // only update pBad if it's nonzero; re-use previous nonzero pBad if the current one is zero.
+        static double pBad;
+        if(trueAcceptingProbability()>0) pBad = trueAcceptingProbability();
 
-			coreCount[source1]++;
-			coreFreq[source1][(*A)[source1]]++;
-			weightedCoreFreq[source1][(*A)[source1]] += pBad;
-			totalCoreWeight[source1] += pBad;
+        coreCount[source1]++;
+        weightedCoreFreq[source1][(*A)[source1]] += pBad;
+        totalCoreWeight[source1] += pBad;
 
-		#endif
-    }
+        coreCount[source2]++;
+        weightedCoreFreq[source2][(*A)[source2]] += pBad;
+        totalCoreWeight[source2] += pBad;
 
+        if(wasBadMove)
```

Comment:
If you create the "worseHole" above, then you can get rid of this if-else statement and just use "bestHole" and "worseHole".